network/p2p/p2pnode/libp2pNode_test.go

```diff
@@ -242,3 +249,168 @@ func TestNode_HasSubscription(t *testing.T) {
 	topic = channels.TopicFromChannel(channels.ConsensusCommittee, unittest.IdentifierFixture())
 	require.False(t, node.HasSubscription(topic))
 }
+
+// TestCreateStream_SinglePairwiseConnection ensures that despite the number of concurrent streams created from peer -> peer only a single
+// connection will ever be created between two peers on initial peer dialing and subsequent streams will reuse that connection.
+func TestCreateStream_SinglePairwiseConnection(t *testing.T) {
+	sporkId := unittest.IdentifierFixture()
+	nodeCount := 3
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	signalerCtx := irrecoverable.NewMockSignalerContext(t, ctx)
+
+	nodes, ids := p2ptest.NodesFixture(t,
+		sporkId,
+		"test_create_stream",
+		nodeCount,
+		p2ptest.WithDefaultResourceManager(),
+		p2ptest.WithPreferredUnicasts(nil))
```

Comment:
Would you please provide some insights on why setting to `nil` is needed?

Reply:
https://github.com/onflow/flow-go/pull/3902/commits/694c35d4921f856e28145e61dacc2f6058f3ce85